calendarsubscription/calendarsubscription.go

```diff
@@ -46,3 +50,45 @@ func (cs CalendarSubscription) Normalize() (*CalendarSubscription, error) {
 
 	return &cs, nil
 }
+
+func (cs CalendarSubscription) renderICalFromShifts(shifts []oncall.Shift, reminderMinutes []int) ([]byte, error) {
+	type iCalOptions struct {
+		Shifts          []oncall.Shift `json:"s,omitempty"`
+		ReminderMinutes []int          `json:"r,omitempty"`
+	}
+
+	var iCalTemplate = `
+		BEGIN:VCALENDAR
+		VERSION:2.0
+		PRODID:-//GoAlert//Calendar Subscriptions//EN
+		CALSCALE:GREGORIAN
+		METHOD:PUBLISH
+		{{range .Shifts}}
+		BEGIN:VEVENT
+		SUMMARY:On-Call Shift
+		DTSTART:{{.Start}}
+		DTEND:{{.End}}
+		END:VEVENT
+		{{end}}
+
+		{{range .ReminderMinutes}}
+		BEGIN:VALARM
+		ACTION:DISPLAY
+		DESCRIPTION:REMINDER
+		TRIGGER:-PT{{.}}M
+		END:VALARM
+		{{end}}
+
+		END:VCALENDAR`
+
+	iCal := template.Must(template.New("iCal").Parse(iCalTemplate))
```

Comment:
Let's do this outside the function as we do in `search.go` and `playground.go`, no need to recompile. Also `Must` will cause a panic at runtime if used like this (and the template was invalid).